src/Google/Ads/GoogleAds/Util/FieldMasks.php

```diff
@@ -105,31 +106,49 @@ public static function allSetFieldsOf(Message $message)
         return self::compare($defaultMessage, $message);
     }
 
+    /**
+     * Returns true if the provided repeated field is null or doesn't have any members.
+     *
+     * @param RepeatedField|null $field the repeated field to check
+     * @return bool true if the field is empty
+     */
+    private static function isEmpty(?RepeatedField $field): bool
+    {
+        return is_null($field) || count($field) === 0;
+    }
+
     /**
      * Builds the paths to the fields that are different between original and modified message.
      *
      * @param array $paths the resulting paths from the computation
      * @param string $currentField the current field name
-     * @param Message $original the original message
-     * @param Message $modified the modified message
+     * @param Message|null $original the original message
+     * @param Message|null $modified the modified message
      */
     private static function buildPaths(
         array &$paths,
         $currentField,
-        Message $original,
-        Message $modified
+        ?Message $original,
+        ?Message $modified
     ) {
-        $descriptor = self::getDescriptorForMessage($original);
+        $descriptor = is_null($original)
+            ? self::getDescriptorForMessage($modified) : self::getDescriptorForMessage($original);
         for ($i = 0; $i < $descriptor->getFieldCount(); $i++) {
             $fieldDescriptor = $descriptor->getField($i);
             $fieldName = self::getFieldName($currentField, $fieldDescriptor);
 
             $getter = Serializer::getGetter($fieldDescriptor->getName());
-            $originalValue = $original->$getter();
+            $originalValue = is_null($original) ? null : $original->$getter();
             $modifiedValue = $modified->$getter();
 
             if (self::isFieldRepeated($fieldDescriptor)) {
-                if ($originalValue != $modifiedValue) {
+                // For protobuf objects, the repeated fields that have no members are semantically
+                // the same as the ones that are null. If both are empty because of any cases, we
+                // will not add their field name to the path, because nothing has changed.
+                if (
+                    !self::isEmpty($originalValue) && !self::isEmpty($modifiedValue)
```

Comment:
> If I understand well, it means that the generated field mask will never include the path of repeated fields that have empty arrays as modified values?

No, if one is empty and another is not, then the field path will still be created.

> If this is the case, I think it is problematic because setting a repeated field to an empty array semantically means "remove all existing items" in the Google Ads API and it will not be covered.

Regarding this, it's never possible with the current protobuf library, because the protobuf lib will automatically create a new repeated field for you when you create a new message.
You cannot distinguish between the auto generated empty repeated field or the intentionally empty repeated field.

We cannot do anything with that right now. Let's talk outside this PR.

Reply:
> > If I understand well, it means that the generated field mask will never include the path of repeated fields that have empty arrays as modified values?
> 
> No, if one is empty and another is not, then the field path will still be created.

I am likely missing something obvious here: this condition checks that both values are not empty so it does not add the field path when the modified value is empty. May be it is added in a different way?

Reply:
Please see the updated version that I uploaded today. :)